### DISCUSSION TITLE
New Monogame/New Developer struggling to get accurate projectiles

### DISCUSSION THREAD

**Discussion by Cev0li** [MonoGame/MonoGame#8552] [Graphics]
Building a little sandbox to code up features for the first time before I use the engine to make my first game. I am struggling to get the projectile to fire on click accurately. The logic is found in Projectile and Player class. It seems like the top left  quadrant is accurate and the accuracy falls off as you click counter clockwise. 

The mentioned classes should be clean enough to navigate/understand quickly. There are some quirks and some areas that need to be cleaned up. Project can be found here: https://github.com/Cev0li/learnMonogame

**Comment by squarebananas** [comment]
This implies `pos` is the projectile's top-left and `_location` is the center:
`_location = new Vector2(pos.X + SpriteTexture.Height / 2, pos.Y + SpriteTexture.Width / 2);`

This implies `pos` is the projectile's center as the origin of the image is in the center (the origin is where the position will be represented in pre-transformed image space):

```
Globals.spriteBatch.Draw(
    _texture,
    new Rectangle((int)_pos.X, (int)_pos.Y, (int)Size.X, (int)Size.Y),
    _sRect,
    Color.White,
    rotation,
    new Vector2(SpriteTexture.Height / 2, SpriteTexture.Width / 2),
    SpriteEffects.None,
    0.0f);
```

<details>
<summary>Try to resolve what <code>pos</code> does so that it represents the same purpose in update and draw. If it is the center then see if your solution matches this:</summary>
Change line 31 to<code>_location = pos;</code>
</details>

**Reply by squarebananas** [reply]
Also your projectiles are square images (16x16), but if they weren't you have width and height the wrong way round.
